### ISSUE TITLE
FIPS 140-2 Enablement

### ISSUE THREAD

**Issue by retr0h** [k3s-io/kine#210]
Any chance of building this project with FIPS 140-2 crypto, such that this could be used with RKE2 in the future?

**Comment by brandond** [comment]
Is there anything here that needs to change? The RKE2 FIPS enablement is 99% on the build side. If we ever to enable kine in RKE2, it would be built in to the main binary in the same way as K3s, and covered by the same processes (goboring, SLE BCI) that provide RKE2 with its FIPS crypto bits.

**Comment by retr0h** [comment]
Given RKE2 [isn't likely to embed Kine](https://github.com/rancher/rke2/discussions/4626), was thinking Kine would need to be built with the FIPS crypto bits, if running along side an RKE2 deployment, and point RKE2 to an external etcd backend (Kine).

**Comment by brandond** [comment]
Yes, at the moment we're unlikely to allow for anything other than the embedded etcd. If we changed that, we would likely embed kine. It's actually already in the binary (inherited from k3s), it's just that we force the use of embedded etcd datastore.

**Comment by dereknola** [comment]
RKE2 now does embed Kine, but not with any FIPS-140-2 Crypto.
